source/particle/property/function.cc

```diff
@@ -0,0 +1,124 @@
+ */
+
+#include <aspect/particle/property/function.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      template <int dim>
+      Function<dim>::Function()
+        :
+        function (1)
+      {}
+
+      template <int dim>
+      void
+      Function<dim>::initialize_one_particle_property(const Point<dim> &position,
+                                                      const Vector<double> &,
+                                                      const std::vector<Tensor<1,dim> > &,
+                                                      std::vector<double> &data) const
+      {
+        data.push_back(function.value(position));
+      }
+
+      template <int dim>
+      std::vector<std::pair<std::string, unsigned int> >
+      Function<dim>::get_property_information() const
+      {
+        const std::vector<std::pair<std::string,unsigned int> > property_information (1,std::make_pair("function",1));
+        return property_information;
+      }
+
+
+      template <int dim>
+      void
+      Function<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.enter_subsection("Function");
+            {
+              Functions::ParsedFunction<dim>::declare_parameters (prm, 1);
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+
+
+      template <int dim>
+      void
+      Function<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.enter_subsection("Function");
+            try
+              {
+                function.parse_parameters (prm);
+              }
+            catch (...)
+              {
+                std::cerr << "ERROR: FunctionParser failed to parse\n"
+                          << "\t'Postprocess.Tracers.Function'\n"
+                          << "with expression\n"
+                          << "\t'" << prm.get("Function expression") << "'";
+                throw;
+              }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      ASPECT_REGISTER_PARTICLE_PROPERTY(Function,
+                                        "function",
+                                        "Implementation of a model in which the tracer "
+                                        "property is given in terms of an explicit formula "
+                                        "that is elaborated in the parameters in section "
+                                        "``Tracers|Function''. The format of these "
```

Comment:
I think you should say that the property is then set to the value you get by *evaluating the function at the initial location of the particle* (I assume that is what it is).